articles/an-introduction-to-graph-neural-network/index.md

```diff
@@ -0,0 +1,188 @@
+The data generated in the above-mentioned fields are represented in a Euclidean space. That is, spaces with a finite number of dimensions i.e. 2D or 3D spaces.
+However, in most current applications, generated data is generated from non-Euclidean domains that represent data as graphs with relationships and mutual dependency between objects. A good example is e-commerce. The interaction between users and products can be exploited to make accurate recommendations. 
+This has led to an increasingly growing interest in deep learning research that focuses on the structure of graph data.
+
+In this article, we provide a comprehensive overview of:
+
+1. What is a Graph?
+
+2. What is Graph Neural Networks?
+
+3. Graph Learning Python Libraries
+
+4. Models of Graph Neural Networks
+
+5. Applications of Graph Neural Networks
+
+
+### Prerequisites
+
+This article assumes a basic understanding of Machine Learning (ML) and Deep Learning (DL). For an introduction to ML and DL, feel free to check out my previous [article](/https://www.section.io/engineering-education/differences-between-artificial-intelligence-machine-learning-and-deep-learning//) 
+
+### What is a Graph?
+
+In Computer Science, Graphs are data structures that model a set of objects and their relationships. These objects represent the nodes and the relationships represent edges. 
+
+![A Graph data structure](/engineering-education/an-introduction-to-graph-neural-network/edges-links.png)
+
+Let's assume a graph, G. This graph can be described by:
+* V as the vertex set 
+* E as the edges
+
+Then,   G = (V,E)
+
+It is important to note that the vertex, V, often refers to the nodes.
+
+Let's take an example of a transportation system. This can be thought of as a graph. The locations in the system represent the vertice, V, and the connections between these locations representing the edges.
+
+### What is Graph Neural Network (GNN)?
+
+Graph Neural Networks is a technique in deep learning which extends existing neural networks for processing data on graphs. 
+
+GNNs motivation stems from three areas of study:
+
+* Convolution Neural Networks (CNN)
+These are deep neural networks used to analyze image data. It is used to solve image processing tasks.
+
+* Recurrent Neural Networks (RNN)
+RNN is a type of neural network that passes information internally from one time step to the next to process a sequence. It's commonly used to solve Natural Language Processing (NLP) and speech recognition tasks.  
+
+* Auto-Encoders (AE)
+This is a type of Artificial Neural Network (ANN). It consists of an encoder, code, and a decoder. It encodes the representations of a set of data. It then decodes the reduced encoded data generating new data similar to that of its original input. 
+
+Based on these three motivations, GNNs can collectively combine information from the graph structure.
+
+Common applications of GNN are in node classification, link prediction, and graph classification. 
```

Comment:
When you mention common, you should ensure the reader knows about the applications. Further details into node classification, link prediction, and graph classification are necessary. An example per each application would be helpful. 

I was kept guessing what nodes would I classify? what links would I predict? If link prediction is a task, how is it different from knowledge graphs? So, the better idea would be to raise questions in the reader's mind and guide them through the answer as well. There lies the value of technical writing. Solving a problem for the reader.